code/sound/sound.cpp

```diff
@@ -655,20 +655,26 @@ sound_handle snd_play_3d(game_snd* gs, vec3d* source_pos, vec3d* listen_pos, flo
 		Int3();
 		return sound_handle::invalid();
 	}
+	if (gs->flags & GAME_SND_NOT_VALID) {
+		return sound_handle::invalid();
+	}
 
 	MONITOR_INC( Num3DSoundsStarted, 1 );
 	
 	auto entry = gamesnd_choose_entry(gs);
 
 	if (!entry->id.isValid()) {
 		entry->id = snd_load(entry, gs->flags);
-		MONITOR_INC( Num3DSoundsLoaded, 1 );
+		MONITOR_INC(Num3DSoundsLoaded, 1);
 	} else if (entry->id_sig != Sounds[entry->id.value()].sig) {
 		entry->id = snd_load(entry, gs->flags);
 	}
 
-	if (!entry->id.isValid())
+	if (!entry->id.isValid()) {
+		Warning(LOCATION, "Failed to load one or more sounds for gamesnd %s!", gs->name.c_str());
+		gs->flags |= GAME_SND_NOT_VALID;
```

Comment:
There is similar code for attempting to load the sound in `snd_play()` and `snd_play_looping()`. We should add the `GAME_SND_NOT_VALID` flag there if needed as well, correct?

In fact, the code fragment starting with `auto entry =` a few lines above is nearly identical in all three `snd_play*()` functions, except that a couple of them call `MONITOR_INC()`.

Perhaps this common code fragment could be refactored into a new function, say
```
game_snd_entry* snd_try_load(game_snd* gs, bool 3d, bool looping)`
```
(or use an `enum` to indicate sound type of 2D vs. 3D vs. 2D looping)

That should ensure that we have to set `GAME_SND_NOT_VALID` in only one place in the code.

Reply:
I've added similar code to the other places, and tweaked them a bit.  I don't really feel like refactoring things now though, and one could make the case to keep this PR simple since it's a bugfix.

Reply:
Thanks, and fair point about not refactoring here. The code duplication it would eliminate wouldn't be much, anyway.